src/main/webapp/progress.js

```diff
@@ -0,0 +1,368 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Function for progess.html, checks what kind of user the viewer is, and loads progress information
+ */
+function loadProgress() {
+    fetch('/login-status').then(response => response.json()).then((loginStatus) => {
+        console.log(loginStatus);    
+        loadProgressHelper(document, loginStatus);
+    });
+}
+
+function loadProgressHelper(document, loginStatus) {
+    getExperiences(document, loginStatus);
+    getGoals(document, loginStatus);
+    getAchievements(document, loginStatus);
+    getPastSessionsAndTopics(document, loginStatus);
+}
+
+/** A function that adds event listeners to a DOM objects. */
+function addEventListeners() {
+    document.getElementById("experiences-form").addEventListener('submit', event => {
+        event.preventDefault();
+        addExperience(window);
+    });
+
+    document.getElementById("goals-form").addEventListener('submit', event => {
+        event.preventDefault();
+        addGoal(window);
+    });
+}
+
+function getExperiences(document, loginStatus) {
+    var queryString = new Array();
+    window.onload = readStudentID(queryString, window);
+    const studentID = queryString["studentID"];
+
+    const params = new URLSearchParams();
+    params.append('studentID', studentID);
+    fetch('/add-experience?studentID=' + studentID, {method: 'GET'}).then(response => response.json()).then((experiences) => {
```

Comment:
Both the tutor and student can view experience/goals right? In that case, we would need to keep this as a url parameter because a user's session only stores the user id of that user.

Reply:
Only the student can add or delete goals/experiences but both student and tutor can view the page. We can fix that after merging then